### DISCUSSION TITLE
Compilation Issues With BEAMnrc

### DISCUSSION THREAD

**Discussion by xiaozhh007** [nrc-cnrc/EGSnrc#1127] [Questions] [answered]
Hi,
I'm compiling beamnrc and it failed after adding a **'pyramids'** structure. After removing it the compilation succeeds again, The structure is used to simulate a wedge plate of Linac, the approximate graphic looks like this:
The XZ plane:
![wedge](https://github.com/nrc-cnrc/EGSnrc/assets/138585202/a6dd286b-772c-450d-82f6-de673e1cb847)
and the YZ plane:
![wedge y](https://github.com/nrc-cnrc/EGSnrc/assets/138585202/2811b436-1752-4603-a63f-8a0c4ff76abe)
![error](https://github.com/nrc-cnrc/EGSnrc/assets/138585202/34e8fb9a-6be7-4c65-984d-a87c4e640c69)
  
**I am wondering if the Pyramids structure is the problem, And if so how to I fix this. Thank you.**

**Comment by xiaozhh007** [comment]
I decreased the number of CMs from 9 to 8, and still get the same error. looked the .F file, found same situation discussed at https://github.com/nrc-cnrc/EGSnrc/issues/734 
.F corrupted lines:

 ZZZZZZZZZ
Z
ZZZZZZZZZZ

"Z

ZZZZZZZZZ
Z&.

but when I simulated single CM with Pyramids structure, the  error was solved. Since the default maximum number of CMs is 25. I'm more curious if there are any other parameters that exceed the limit.

**Reply by rtownson** [reply, marked as answer]
Any time you change the number of CMs, you should create an entirely new accelerator with a new name. If you are changing the .module file, and then re-building and re-compiling the same accelerator, I could imagine there might be issues like this. Always compile the accelerator before you start creating an input file (and then you never need to compile it again).

To be clear, my suggestion is to try again with a completely new accelerator name so that a new BEAM_myAccelerator folder is created.

**Reply by xiaozhh007** [reply]
Thank you for your reply，After checking the module file rename it  and recompiling, it works. The weird thing is that I didn't change much of the contents, but only removed some CMs' comment underscore "_".
